test/unit/features/options/mixins.spec.js

```diff
@@ -109,4 +109,32 @@ describe('Options mixins', () => {
     expect(vm.b).toBeDefined()
     expect(vm.$options.directives.c).toBeDefined()
   })
+
+  it('should not mix global mixined lifecycle hook twice', () => {
+    const spy = jasmine.createSpy('global mixed in lifecycle hook')
+    Vue.mixin({
+      created() {
+        spy()
+      }
+    })
+
+    const mixin1 = Vue.extend({
+      methods: {
+        a() {}
+      }
+    })
+
+    const mixin2 = Vue.extend({
+      mixins: [mixin1],
+    })
+
+    const Child = Vue.extend({
+      mixins: [mixin2],
+    })
+
+    const vm = new Child()
+
+    expect(typeof vm.$options.methods.a).toBe('function')
```

Comment:
what is this test for?

Reply:
This test is to make sure the fix does not break normal inheritance from grandparent, in this test, `mixin1` and `Child`